storage/grpc_client.go

```diff
@@ -1492,10 +1551,144 @@ func (r *gRPCReader) recv() (*storagepb.ReadObjectResponse, error) {
 		// reopen the stream, but will backoff if further attempts are necessary.
 		// Reopening the stream Recvs the first message, so if retrying is
 		// successful, the next logical chunk will be returned.
-		msg, err = r.reopenStream()
+		msg, err := r.reopenStream()
+		return msg.GetChecksummedData().GetContent(), err
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return readObjectResponseContent(r.databuf)
+}
+
+// readObjectResponseContent returns the checksummed_data.content field of a
+// ReadObjectResponse message, or an error if the message is invalid.
+// This can be used on recvs of objects after the first recv, since only the
+// first message will contain non-data fields.
+func readObjectResponseContent(b []byte) ([]byte, error) {
+	const (
+		readObjectResponse_checksummedData = protowire.Number(1)
+		checksummedData_content            = protowire.Number(1)
+	)
+
+	checksummedData, err := readProtoBytes(b, readObjectResponse_checksummedData)
+	if err != nil {
+		return b, fmt.Errorf("invalid ReadObjectResponse: %v", err)
+	}
+	content, err := readProtoBytes(checksummedData, checksummedData_content)
+	if err != nil {
+		return content, fmt.Errorf("invalid ReadObjectResponse: %v", err)
+	}
+
+	return content, nil
+}
+
+// readFullObjectResponse returns the ReadObjectResponse encoded in the
+// wire-encoded message buffer b, or an error if the message is invalid.
+// This is used on the first recv of an object as it may contain all fields of
+// ReadObjectResponse.
```

Comment:
1. Is there additional overhead of reading the full response vs. just the data? 2. Do we have a guarantee that the API will _never_ return these fields on later responses?

Reply:
1. I don't think there will be much overhead of reading the full response unless it does return the metadata; however:
2. Even if the API does return these fields, we don't use them at all. All we did previously is return the content in `Read`, the reference for anything else is discarded when Read returns. Currently the `recv` doesn't even return the rest of the msg either.

Reply:
Thinking about it, the API may return at least the ChecksummedData.Crc32C field in every message (since this is relevant per msg). If we add checks on every message, we'd need to grab that too, at least. However, as said, we aren't currently using anything other than the content.

Reply:
Got it, that sounds reasonable.